### DISCUSSION TITLE
kill @@$generichide and @@$genericblock pls! or at least allow to globally ignore those

### DISCUSSION THREAD

**Discussion by ghost** [uBlockOrigin/uBlock-issues#2138] [General]
<details><summary>Prerequisites</summary>

### Prerequisites

- [X] I verified that this is not a filter issue (MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues))
- [X] This is not a [support issue or a question](https://old.reddit.com/r/uBlockOrigin/)
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue
- [X] The issue is not present after wholly disabling uBlock Origin ("uBO") in the browser
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### I tried to reproduce the issue when...

- [X] uBO is the only extension
- [X] uBO with default lists/settings
- [X] using a new, unmodified browser profile

</details>

### Description

<img width="651" alt="2022-06-11_015017" src="https://user-images.githubusercontent.com/415238/173160262-e4e44204-a0d6-4fd7-a7eb-d46ab7be712c.png">

I'm starting to see trend in use of `$generichide` and `$genericblock` again.

it breaks a lot of good old, generic web-cleaning rules.

every d.bag only needs to add `@@||d.bag.com^$genericblock`  
and `@@||d.bag.com^$generichide`  
to immediately and effectively disable a lot of rules with very little effort.

it is pretty much just like saying: oh? you've got some rules? na, I don't like that, let me show you some ads regardless.  

EasyList would have <del>quite a lot</del> \*some\*&nbsp;<sup>(<a href="https://github.com/uBlockOrigin/uBlock-issues/discussions/2138#discussioncomment-2931408">edit</a>)</sup> of those.
and a lot of smaller lists would copy from one to another,  
spreading this cr^ap around.

this is essentially abusing the `@@` syntax,  
which was meant to unbreak site's functionality,  
it become a kind of a master-switch,  
being such a major game changer,  
I would very much like (pls) a way to integrate a master switch of our own,  
allowing us to prevent that syntax abuse..  

...feel free to discuss/re-define this better if you have a better experience with those stuff :|

Ps.
allowing an even stronger override, such as:  
- `*$important,generichide`
- `*$important,genericblock`
would be hilarious!

thanks!

### A specific URL where the issue occurs

https://blog.adblockplus.org/development-builds/new-filter-options-generichide-and-genericblock

### Steps to Reproduce

0. varies with lists.

### Expected behavior

ability to override, ..or something..

### Actual behavior

1. being d.bag advertiser.
2. $$$ -> list maintainer: += '@@d.bag.com$genericblock' + '@@d.bag.com$generichide'
3. profit.

<details><summary>versions</summary>

### uBlock Origin version

recent one.

### Browser name and version

Chromium

### Operating System and version

Kali Linux
</details>

**Comment by gorhill** [comment]
uBO does not support `genericblock`, and you can already use `*$important,generichide` if you want.

---

If you do not trust the list maintainers, you are free to not use their lists, you can maintain your own. It makes no sense to not trust list maintainers for a specific list and yet use this list.

**Comment by ryanbr** [comment]
> EasyList and even AdGuard Base would have quite a lot of those.

We have only one genericblock; `easylist/easylist_specific_block.txt:@@||spiegel.de^$genericblock,generichide`

And we generichide for webcompat purposes, Where domains aren't serving ads. https://github.com/easylist/easylist/blob/master/easylist/easylist_allowlist.txt#L298

and some previous anti-ad cosmetic checks: https://github.com/easylist/easylist/blob/master/easylist/easylist_allowlist.txt#L379

Nothing "suspect" or anything, given you can look through all our commits, pull requests etc. We'll occasionally prune these when the sites remove the anti-ad cosmetic checks. I'll be happy to re-review these ghides later in the week

**Reply by iam-py-test** [reply]
Maybe I am wrong, but can't you [badfilter](https://github.com/uBlockOrigin/uBlock-issues/wiki/Static-filter-syntax#badfilter) the offending rule? (or, if that fails, do something like this `[domain]##table,thead,tbody,tr,td,th:style(border-color:rgba(150,150,150,.30) !important; border-style:dashed !important; border-width:1px !important; border-collapse:collapse !important; border-spacing:0px !important;)`)
Not a pretty solution, but it works.

**Reply by JohnyP36** [reply]
Instead of using `generichide` to pass adblock detection it's better to make a specific rule for that domain allowing the request or element that causes detection.
